### DISCUSSION TITLE
关于README中“请注意”内容的疑问？

### DISCUSSION THREAD

**Discussion by scottcgi** [ldqk/Masuit.Tools#112] [General]
> 一旦使用本开源项目以及引用了本项目或包含本项目代码的公司因为违反劳动法（包括但不限定非法裁员、超时用工、雇佣童工等）在任何法律诉讼中败诉的，一经发现，本项目作者有权利追讨本项目的使用费（公司工商注册信息认缴金额的2-5倍作为本项目的授权费），或者直接不允许使用任何包含本项目的源代码！ 人力外包公司或 007公司需要使用本类库，请联系作者进行商业授权！其他企业或个人可随意使用不受限。007那叫用人，也是废人。8小时工作制才可以让你有时间自我提升，将来有竞争力。反对007，人人有责！——引自[README-请注意](https://github.com/ldqk/Masuit.Tools?tab=readme-ov-file#%E8%AF%B7%E6%B3%A8%E6%84%8F)

1. 此约束是否包含在[LICENSE](https://github.com/ldqk/Masuit.Tools/blob/master/LICENSE)中，成为许可约束的一部分？
2. 如果非中文用户，只看到了“MIT License”，是否需要遵守此约束？
3. 此约束是否只针对中文用户？
4. 如果使用此项目的是员工的“个人行为”，那么出现符合上述违约，“使用费”是向公司索取，还是个人？
5. 如果员工个人使用，公司并不知情，然后产生劳动纠纷，员工主动举报违约使用此项目，这种情况是否“追讨使用费”？
6. 如何有效发现，违反劳动法的案件中，使用了这个项目？——如果不能有效监督，此约束可能会被轻视或无视。
7. 如果007公司，支付商业授权费，是否就可以合理使用这个项目，如此即是，有钱公司就可以在007的情况下，付费合理使用此项目？
8. 在007公司的员工，已经很辛苦了，但有时候是不得不的现实，结果想使用此项目减轻劳动负担，单遇到此约束该如何是好？——007公司如果不愿付费获得授权，自然就会逼迫员工自己加班实现解决方案，此时约束没有影响到公司而是个人。

**Comment by ldqk** [comment]
非常感谢您的提问，我将针对您的提问逐一解答，首先可以明确的是，**此补充条款，针对的是公司而非个人**；
Q1. 此约束是否包含在[LICENSE](https://github.com/ldqk/Masuit.Tools/blob/master/LICENSE)中，成为许可约束的一部分？
A：此补充约束不包含在主LICENSE中，仅为使用本项目时针对MIT License的补充约束，因为此补充约束不对所有地区的用户生效。

Q2. 如果非中文用户，只看到了“MIT License”，是否需要遵守此约束？
A：非中文用户无需遵守此补充约束。

Q3. 此约束是否只针对中文用户？
A：此约束只针对中国地区的中文用户生效。

Q4. 如果使用此项目的是员工的“个人行为”，那么出现符合上述违约，“使用费”是向公司索取，还是个人？
A：如果因为员工的“个人行为”造成违规使用的情况，“使用费”依然是向公司索取，因为使用此项目的员工依然是服务于公司的项目。

Q5. 如果员工个人使用，公司并不知情，然后产生劳动纠纷，员工主动举报违约使用此项目，这种情况是否“追讨使用费”？
A：员工个人使用，公司并不知情，出现这种情况，说明公司在项目管理的流程上是极其不规范的（为什么不做技术选型评估？为什么不做Code Review？为什么不做代码安全扫描？），因违反劳动法员工主动举报违约使用此项目成立的，依然会向其公司追讨使用费，公司应当为自己管理的不规范付出应有的代价。

Q6. 如何有效发现，违反劳动法的案件中，使用了这个项目？——如果不能有效监督，此约束可能会被轻视或无视。
A：违规使用本项目的，目前只能通过违反劳动法的案件中关联的员工监督并向本项目所有者举报投诉，留存证据。

Q7. 如果007公司，支付商业授权费，是否就可以合理使用这个项目，如此即是，有钱公司就可以在007的情况下，付费合理使用此项目？
A：本项目不支持主动支付商业授权费，007公司应该考虑增加更多的就业岗位，提升员工的工作环境，以恢复965这种工作作息。

Q8. 在007公司的员工，已经很辛苦了，但有时候是不得不的现实，结果想使用此项目减轻劳动负担，单遇到此约束该如何是好？——007公司如果不愿付费获得授权，自然就会逼迫员工自己加班实现解决方案，此时约束没有影响到公司而是个人。
A：007公司的员工，辛苦是必然的，为什么会辛苦？还不是因为公司无尽的压榨才导致的007（拿着965的工资，干着007的活），公司为什么不考虑多增加一些就业岗位，减少007这种情况的存在？！减轻劳动负担的正确途径是增加人手，这样相同的工作量有更多的人来均摊，才能有效减轻工作负担，单靠一个效率工具库是不能减轻劳动负担的，甚至因为一个效率工具库，工作量是减少了，那么公司就有更多的后续工作量来提前压到员工身上，这对员工是不利的，当然，007公司不一定使用本项目就是违规的，如果007公司能够严格按照劳动法去**支付员工在标准工时以外的加班费**，依然是可以无限制使用的。
